src/java/org/dom4j/io/XMPPPacketReader.java

```diff
@@ -381,9 +390,13 @@ public Document parseDocument() throws DocumentException, IOException, XmlPullPa
                     int nsStart = pp.getNamespaceCount(pp.getDepth() - 1);
                     int nsEnd = pp.getNamespaceCount(pp.getDepth());
                     for (int i = nsStart; i < nsEnd; i++) {
-                        if (pp.getNamespacePrefix(i) != null) {
-                            newElement
-                                    .addNamespace(pp.getNamespacePrefix(i), pp.getNamespaceUri(i));
+                        final String namespacePrefix = pp.getNamespacePrefix( i );
+                        final String namespaceUri = pp.getNamespaceUri( i );
+                        if ( namespacePrefix != null ) {
+                            newElement.addNamespace( namespacePrefix, namespaceUri );
+                        } else if ( !( pp.getDepth() <= 2 && IGNORED_NAMESPACE_ON_STANZA.contains(namespaceUri) ) ) {
+                            // Do not include certain default namespace on the root-element ('stream') or stanza level. This makes stanzas re-usable between, for example, c2s and s2s.
+                            newElement.addNamespace( "", namespaceUri );

```

Comment:
So I think this won't catch:

`<message xmlns='jabber:client'><body xmlns='jabber:client'>Blah</body></message>`

This probably doesn't matter for now.

Reply:
Should we even try to normalize XML that passes through?